x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/HttpCertificateCommand.java

```diff
@@ -0,0 +1,1149 @@
+            terminal.println(Terminal.Verbosity.VERBOSE, "\tName: " + cert.name);
+            terminal.println(Terminal.Verbosity.VERBOSE, "\tSubject: " + cert.subject);
+            terminal.println(Terminal.Verbosity.VERBOSE, "\tDNS Names: " + Strings.collectionToCommaDelimitedString(cert.dnsNames));
+            terminal.println(Terminal.Verbosity.VERBOSE, "\tIP Names: " + Strings.collectionToCommaDelimitedString(cert.ipNames));
+            terminal.println(Terminal.Verbosity.VERBOSE, "\tKey Size: " + cert.keySize);
+            terminal.println(Terminal.Verbosity.VERBOSE, "\tValidity: " + toString(cert.validity));
+            certificates.add(cert);
+
+            if (multipleCertificates && terminal.promptYesNo("Generate additional certificates?", true)) {
+                nodeDescription = "node #" + (certificates.size() + 1);
+                continue;
+            } else {
+                break;
+            }
+        }
+
+        printHeader("What password do you want for your private keys?", terminal);
+        char[] password;
+        if (csr) {
+            terminal.println("Your private keys will be stored as a PEM formatted file.");
+            terminal.println("We recommend that you protect your private keys with a password");
+            terminal.println("");
+            terminal.println("If you do not wish to use a password, simply press <enter> at the prompt below.");
+            password = readPassword(terminal, "Provide a password for the private key: ", true);
+        } else {
+            terminal.println("Your private keys will be stored in a PKCS#12 keystore file named \"http.p12\".");
+            terminal.println("This type of keystore is always password protected, but it is possible to use a");
+            terminal.println("blank password.");
+            terminal.println("");
+            terminal.println("If you wish to use a blank password, simply press <enter> at the prompt below.");
+            password = readPassword(terminal, "Provide a password for the \"http.p12\" file: ", true);
+        }
+
+        printHeader("Where should we save the generated files?", terminal);
+        if (csr) {
+            terminal.println("A number of files will be generated including your private key(s),");
+            terminal.println("certificate request(s), and sample configuration options for Elastic Stack products.");
+        } else {
+            terminal.println("A number of files will be generated including your private key(s),");
+            terminal.println("public certificate(s), and sample configuration options for Elastic Stack products.");
+        }
+        terminal.println("");
+        terminal.println("These files will be included in a single zip archive.");
+        terminal.println("");
+        Path output = resolvePath("elasticsearch-ssl-http.zip");
+        output = tryReadInput(terminal, "Where should we save the output?", output, this::resolvePath);
```

Comment:
NIT: We could rephrase that to "what should the filename of the output file be" or something similar, because the way this reads it could also meant that we ask for the directory where to store the `elasticsearch-ssl-http.zip` file.